Formula/hasura-cli.rb

```diff
@@ -0,0 +1,34 @@
+class HasuraCli < Formula
+  desc "Command-Line Interface for Hasura GraphQL Engine"
+  homepage "https://hasura.io"
+  url "https://github.com/hasura/graphql-engine/archive/v1.2.2.tar.gz"
+  sha256 "59e15fe760794bb0ec69b62d0bf13d7925c2f595d78ee4121fc82fa67c36dbf9"
+
+  depends_on "go@1.13" => :build
+  depends_on "gox" => :build
+
+  def install
+    cd "cli" do
+      system "gox",
+        "-ldflags",
+          "-X github.com/hasura/graphql-engine/cli/version.BuildVersion=#{version} " \
+          "-X github.com/hasura/graphql-engine/cli/plugins.IndexBranchRef=master -s -w -extldflags '-static'",
+        "-os", "darwin",
+        "-arch", "amd64",
+        "-output", bin/"hasura",
+        "./cmd/hasura/"
+
+      prefix.install_metafiles
+
+      system bin/"hasura", "completion", "bash", "--file", "completion_bash"
+      bash_completion.install "completion_bash" => "hasura"
+
+      system bin/"hasura", "completion", "zsh", "--file", "completion_zsh"
+      zsh_completion.install "completion_zsh" => "_hasura"
+    end
+  end
+
+  test do
+    system "#{bin}/hasura"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.